app/src/main/java/org/mozilla/focus/fragment/DownloadDialogFragment.java

```diff
@@ -78,7 +78,11 @@ private void setButtonOnClickListener(Button button, final Download pendingDownl
                     @Override
                     public void onClick(View v) {
                         sendDownloadDialogButtonClicked(pendingDownload, shouldDownload);
-                        TelemetryWrapper.downloadDialogDownloadEvent();
+                        if (shouldDownload) {
+                            TelemetryWrapper.downloadDialogDownloadEvent();
+                        } else {
+                            TelemetryWrapper.downloadDialogCancelEvent();
+                        }
```

Comment:
Another option here would be to pass "shouldDownload" to TelemetryWrapper.downloadDialogDownloadEvent() and decide what event to build/send in there. Just for making our main code simpler I'd try to reduce telemetry to one line as much as possible. :)